crates/oro-tree/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+use memmap::MmapOptions;
+use semver::{Version, VersionReq};
+use serde::{
+    de::{Deserializer, Error as SerdeError},
+    Deserialize,
+};
+use ssri::Integrity;
+use std::fs::File;
+use std::{collections::HashMap, path::Path};
+use thiserror::Error;
+
+fn default_as_false() -> bool {
+    false
+}
+
+fn parse_integrity<'de, D>(deserializer: D) -> Result<Integrity, D::Error>
+where
```

Comment:
You shouldn't need this. `Integrity` already implements `Deserialize`.

Reply:
It does, but only as an array structure from what I can see?
https://github.com/zkat/ssri-rs/blob/latest/src/integrity.rs#L30

Reply:
uggghhhh, you're right. It has to be manually parsed/stringified. Maybe ssri should implement a custom one that does to_string() and .parse() since that's the "proper" serialization. This works, ig.